ansible/examples/slave_libvirt.yml

```diff
@@ -0,0 +1,235 @@
+---
+
+- hosts: all
+  sudo: true
+  user: ubuntu
+  vars:
+   - jenkins_user: 'jenkins-build'
+   # jenkins API credentials:
+   - api_user: 'ceph-jenkins'
+   - token: '{{ token }}'
+   - api_uri: 'https://jenkins.ceph.com'
+   - nodename: '{{ nodename }}'
+   - labels: '{{ labels }}'
+   - use_jnlp: false
+
+  tasks:
+
+    # vagrant doesn't have repositories, this chacra repo will be better to have
+    # around and can get updates as soon as a new vagrant version is published via
+    # chacractl
+    - name: add the vagrant repository
+      sudo: yes
+      apt_repository:
+        repo: "deb [trusted=yes] https://chacra.ceph.com/r/vagrant/latest/HEAD/ubuntu/xenial/flavors/default/ main"
+        state: present
+        filename: "vagrant"
+
+    # Run the equivalent of "apt-get update" as a separate step
+    - apt: update_cache=yes
+      when: ansible_pkg_mgr  == "apt"
+
+    - name: Install DEB requirements
+      sudo: yes
+      apt: name={{ item }} state=present
+      with_items:
+        - git
+        - python-dev
+        - python-pip
+        - python-virtualenv
+        - libtool
+        - debian-archive-keyring
+        # jenkins-job-builder job:
+        - libyaml-dev
+        - libvirt
+        - qemu-kvm
+        - libvirt-bin
+        - vagrant
+      when: ansible_pkg_mgr  == "apt"
+
+    - name: install the vagrant-libvirt plugin
+      shell: vagrant plugin install vagrant-libvirt
```

Comment:
I think you need to move this task down after the task that creates the ``jenkins_user``.